### PR TITLE
Talisman html reporting support

### PR DESCRIPTION
**Pull request by jaydeepc**

### CODE REVIEW

**Comment by svishwanath-tw** [comment]
$ ./talisman_darwin_amd64 --htmlreport
Please wait while talisman scans entire repository including the git history...
2019/05/18 23:00:18 Cannot create report.json fileopen talisman_html_report/data/report.json: no such file or directory

File: report/report.go [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
-	if err != nil {
-		log.Fatal("Cannot create report.html file", err)
-	}
-	reportTemplate.ExecuteTemplate(htmlFile, "report", r)
-	htmlFile.Close()
+
+	usr, err := user.Current()
+	home_dir = usr.HomeDir
+
+	base_report_dir_path = filepath.Join(home_dir, ".talisman/bin/talisman_html_report")
```
[the remark this answers is not in the document]
Comment (jaydeepc) [reply report/report.go]: Done

File: report/report.go [hunk 2]
```diff
@@ -1,39 +1,49 @@
...
+
+	os.MkdirAll(path, 0755)
+	Dir(base_report_dir_path, "talisman_html_report")
```
[the remark this answers is not in the document]
Comment (jaydeepc) [reply report/report.go]: Done

**Comment by jaydeepc** [comment]
Worked on the comments. But could not reproduce the last one 
```
$ ./talisman_darwin_amd64 --htmlreport
Please wait while talisman scans entire repository including the git history...
2019/05/18 23:00:18 Cannot create report.json fileopen talisman_html_report/data/report.json: no such file or directory

```

From which directory you tried? I tried with 

``` talisman --htmlreport ```

and also 

``` in-Jaydeepc:test-tali jaydeepc$ ~/.talisman/bin/talisman_darwin_amd64 --htmlreport ```

And both worked

**Comment by svishwanath-tw** [comment]
> Worked on the comments. But could not reproduce the last one
> 
> ```
> $ ./talisman_darwin_amd64 --htmlreport
> Please wait while talisman scans entire repository including the git history...
> 2019/05/18 23:00:18 Cannot create report.json fileopen talisman_html_report/data/report.json: no such file or directory
> ```
> 
> From which directory you tried? I tried with
> 
> `talisman --htmlreport`
> 
> and also
> 
> `in-Jaydeepc:test-tali jaydeepc$ ~/.talisman/bin/talisman_darwin_amd64 --htmlreport`
> 
> And both worked

I tried it from talisman directory

**Comment by jaydeepc** [comment]
@svishwanath-tw as discussed, will be making the below changes:
1. use soft links instead of `mv`
Eg: ln `-svf ~/.talisman/html-report-version ~/.talisman/html-report`
instead of `mv html-report-version html-report`

2. Show an error and prompt user to install `html_report` or ask user to just run `--scan` which will give user the JSON report.

3. along with one, also give a message in talisman --help about when to use html report

4.  extract to` ~/.talisman` instead of `~/.talisman/bin`. `bin` is supposed to be for executables

**Comment by jaydeepc** [comment]
@svishwanath-tw , modified the code, please review

**Comment by svishwanath-tw** [comment]
Nice UI work JD. I think the error message still says --htmlreport instead of --scanWithHtml.
Everything else LGTM

File: report/report.go [hunk 3]
```diff
@@ -50,290 +62,15 @@ func GenerateReport(r *detector.DetectionResults, directory string) string {
...
-			<footer class="site-footer">
-  				<div class="wrapper">
-    				<h3 class="footer-heading"><a href="https://github.com/thoughtworks/talisman/">GitHub</a></h3>
-    				<h2 class="footer-heading">&#9400; 2016&nbsp;ThoughtWorks, Inc.</h2>
-  				</div>
-			</footer>
-		</body>
-	</html>`
+func generateErrorMsg() {
+	color.HiMagenta("\nLooks like you are using 'talisman --htmlreport' for scanning.")
```
Comment (svishwanath-tw) [inline report/report.go]: this should say --scanWithHtml instead of --htmlReport

Comment (jaydeepc) [reply report/report.go]: Done

**Pull request merged**